packages/edit-post/src/store/actions.js

```diff
@@ -3,29 +3,34 @@
  */
 import { castArray } from 'lodash';
 
+/**
+ * WordPress dependencies
+ */
+import { dispatch } from '@wordpress/data-controls';
+
 /**
  * Returns an action object used in signalling that the user opened an editor sidebar.
  *
- * @param {string} name Sidebar name to be opened.
+ * @param {?string} name Sidebar name to be opened.
  *
- * @return {Object} Action object.
+ * @yield {Object} Action object.
  */
-export function openGeneralSidebar( name ) {
-	return {
-		type: 'OPEN_GENERAL_SIDEBAR',
-		name,
-	};
+export function* openGeneralSidebar( name ) {
+	yield dispatch(
+		'core/interface',
+		'setSingleActiveArea',
+		'edit-post/complementary-area',
+		name
+	);
 }
 
 /**
  * Returns an action object signalling that the user closed the sidebar.
  *
- * @return {Object} Action object.
+ * @yield {Object} Action object.
  */
-export function closeGeneralSidebar() {
-	return {
-		type: 'CLOSE_GENERAL_SIDEBAR',
-	};
+export function* closeGeneralSidebar() {
+	yield* openGeneralSidebar();
```

Comment:
Is it a bug, why would you call open to close?

Reply:
It is not a bug we are calling openGeneralSidebar without any sidebar name so the name is undefined and in that case, we will make sure no sidebar is available in that area.
But I will update the code, to not use openGeneralSidebar as it is confusing.